src/gui/modalMenu.cpp

```diff
@@ -194,41 +193,51 @@ bool GUIModalMenu::simulateMouseEvent(
 	default:
 		return false;
 	}
-	if (preprocessEvent(mouse_event))
-		return true;
-	if (!target)
-		return false;
-	return target->OnEvent(mouse_event);
+
+	bool retval;
+	m_simulated_mouse = true;
+	do {
+		if (preprocessEvent(mouse_event)) {
+			retval = true;
+			break;
+		}
+		if (!target) {
+			retval = false;
+			break;
+		}
+		retval = target->OnEvent(mouse_event);
+	} while (false);
+	m_simulated_mouse = false;
+
+	return retval;
 }
 
 void GUIModalMenu::enter(gui::IGUIElement *hovered)
 {
 	if (!hovered)
 		return;
-	sanity_check(!m_hovered);
-	m_hovered.grab(hovered);
+	sanity_check(!m_touch_hovered);
+	m_touch_hovered.grab(hovered);
 	SEvent gui_event{};
 	gui_event.EventType = EET_GUI_EVENT;
-	gui_event.GUIEvent.Caller = m_hovered.get();
-	gui_event.GUIEvent.EventType = EGET_ELEMENT_HOVERED;
+	gui_event.GUIEvent.Caller = m_touch_hovered.get();
+	gui_event.GUIEvent.EventType = irr::gui::EGET_ELEMENT_HOVERED;
```

Comment:
`irr::` is never necessary, we have an `using namespace irr;` somewhere

Reply:
Indeed. I tried removing `gui::` too, got compilation errors.